NPE for smt-reachability

batfish> init-testrig test_rigs/example
....
batfish> get smt-reachability 
Status: FAILURE
Question: reachability actions=[ACCEPT] | dstIps=[2.1.2.1]
org.batfish.common.BatfishException: Failed to answer question
   at org.batfish.main.Batfish.answer(Batfish.java:542)
   at org.batfish.main.Batfish.run(Batfish.java:3957)
   at org.batfish.main.Driver$1.run(Driver.java:327)
Caused by: java.lang.NullPointerException
   at com.microsoft.z3.Context.checkContextMatch(Context.java:3882)
   at com.microsoft.z3.Context.mkEq(Context.java:680)
   at org.batfish.smt.PropertyAdder.instrumentReachability(PropertyAdder.java:73)
   at org.batfish.smt.PropertyChecker.allReachable(PropertyChecker.java:64)
   at org.batfish.smt.PropertyChecker.computeReachability(PropertyChecker.java:109)
   at org.batfish.main.Batfish.smtReachability(Batfish.java:4597)
   at org.batfish.question.SmtReachabilityQuestionPlugin$ReachabilityAnswerer.answer(SmtReachabilityQuestionPlugin.java:30)
   at org.batfish.main.Batfish.answer(Batfish.java:539)
   ... 2 more
Summary: 0 (non-assertion) results;

Another command, which seems to fail in a different place:

batfish> get smt-reachability finalNodeRegex="as2border1", finalIfaceRegex="Loopback.*"
Status: FAILURE
Question: reachability actions=[ACCEPT] | dstIps=[2.1.1.1]
org.batfish.common.BatfishException: Failed to answer question
   at org.batfish.main.Batfish.answer(Batfish.java:542)
   at org.batfish.main.Batfish.run(Batfish.java:3957)
   at org.batfish.main.Driver$1.run(Driver.java:327)
Caused by: java.lang.NullPointerException
   at org.batfish.smt.Encoder.lambda$null$9(Encoder.java:599)
   at java.util.HashMap.forEach(HashMap.java:1288)
   at org.batfish.smt.Encoder.lambda$buildCounterExample$10(Encoder.java:597)
   at java.util.HashMap.forEach(HashMap.java:1288)
   at org.batfish.smt.Encoder.buildCounterExample(Encoder.java:592)
   at org.batfish.smt.Encoder.verify(Encoder.java:789)
   at org.batfish.smt.PropertyChecker.computeReachability(PropertyChecker.java:143)
   at org.batfish.main.Batfish.smtReachability(Batfish.java:4597)
   at org.batfish.question.SmtReachabilityQuestionPlugin$ReachabilityAnswerer.answer(SmtReachabilityQuestionPlugin.java:30)
   at org.batfish.main.Batfish.answer(Batfish.java:539)
   ... 2 more
Summary: 0 (non-assertion) results;

--- a/projects/batfish/src/main/java/org/batfish/smt/Encoder.java
+++ b/projects/batfish/src/main/java/org/batfish/smt/Encoder.java
@@ -596,7 +596,7 @@ public class Encoder {
                   .getEnvironmentVars()
                   .forEach(
                       (lge, r) -> {
-                        if (valuation.get(r.getPermitted()).equals("true")) {
+                        if ("true".equals(valuation.get(r.getPermitted()))) {
                           SortedMap<String, String> recordMap = new TreeMap<>();
                           GraphEdge ge = lge.getEdge();
                           String nodeIface =

--- a/projects/batfish/src/main/java/org/batfish/smt/utils/PatternUtils.java
+++ b/projects/batfish/src/main/java/org/batfish/smt/utils/PatternUtils.java
@@ -46,12 +46,14 @@ public class PatternUtils {
               Matcher m2 = p2.matcher(router);
               if (m1.matches() && !m2.matches()) {
                 for (GraphEdge edge : edges) {
-                  Interface i = edge.getStart();
-                  String ifaceName = i.getName();
-                  Matcher m3 = p3.matcher(ifaceName);
-                  Matcher m4 = p4.matcher(ifaceName);
-                  if (m3.matches() && !m4.matches()) {
-                    acc.add(edge);
+                  if (!edge.isAbstract()) {
+                    Interface i = edge.getStart();
+                    String ifaceName = i.getName();
+                    Matcher m3 = p3.matcher(ifaceName);
+                    Matcher m4 = p4.matcher(ifaceName);
+                    if (m3.matches() && !m4.matches()) {
+                      acc.add(edge);
+                    }
                   }
                 }
               }

